### DISCUSSION TITLE
Famicom 15pin cable build instruction issue

### DISCUSSION THREAD

**Discussion by tawyy** [darthcloud/BlueRetro#762] [General]
Hello,
A mistake is on the 15 pin connector. P1-D0 is nowhere.
I think the D signals have bad labels. D0 ~ D4 needs to be renamed to D0 ~ D2 maybe ?
I found the pinout on another website to cross the pinout from NES to 15pin famicom, and it worked.

https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-Cables-Build-Instructions#fc--nes-adapter-cable

**Comment by darthcloud** [comment]
For original Famicom D0 is available only internally on the hard wired controllers, the signal to use on the DB15 is D1.

My pinout is base on official Nintendo schematic.

**Comment by SonnyJim** [comment]
Ah ok, because I was being super confused by this:

https://github.com/darthcloud/BlueRetro/blob/b7b72e0109d431483c66e97ec3b26bf1c78c9d92/main/wired/npiso_io.c#L24

P1_D0 is allocated a GPIO which was confusing me as I couldn't find it anywhere on the DB15.  Am I right in thinking that most Famicom software will allow player 1 input from the expansion port?

**Reply by darthcloud** [reply]
Yes they they almost all support input from both D0 and D1.

I made patch for the few that doesn't:
https://github.com/darthcloud/famicom-exp-patches
